eland/operations.py

```diff
@@ -472,13 +495,26 @@ def _unpack_metric_aggs(
 
                     # Pull multiple values from 'percentiles' result.
                     if es_agg[0] == "percentiles":
-                        agg_value = agg_value["values"]
-
-                    agg_value = agg_value[es_agg[1]]
-
+                        agg_value = agg_value["values"]  # Returns dictionary
+                        if pd_agg == "median":
+                            agg_value = agg_value["50.0"]
+                        # Currently Pandas does the same
+                        # If we call quantile it returns the same result as of median.
+                        elif pd_agg == "quantile" and is_dataframe_agg:
+                            agg_value = agg_value["50.0"]
+                        else:
+                            # We have to filter out the `_as_string` from results
+                            # e.g. 'Cancelled_50.0': 0.0, 'Cancelled_50.0_as_string': 'false'
+                            percentile_values = [
```

Comment:
Does order matter for these values? If so we might need to do some ordering of `agg_value.items()`? if not it's all good.

Reply:
Good call, I forgot that scenario..